src/main/java/com/synopsys/integration/alert/web/user/UserActions.java

```diff
@@ -84,17 +84,41 @@ public UserConfig createUser(UserConfig userConfig) throws AlertDatabaseConstrai
         if (!fieldErrors.isEmpty()) {
             throw new AlertFieldException(fieldErrors);
         }
-        Set<String> configuredRoleNames = userConfig.getRoleNames();
-        Collection<UserRoleModel> roleNames = authorizationUtility.getRoles().stream()
-                                                  .filter(role -> configuredRoleNames.contains(role.getName()))
-                                                  .collect(Collectors.toList());
+
         UserModel userModel = userAccessor.addUser(userName, password, emailAddress);
         Long userId = userModel.getId();
-        authorizationUtility.updateUserRoles(userId, roleNames);
+        Set<String> configuredRoleNames = userConfig.getRoleNames();
+        if (null != configuredRoleNames && !configuredRoleNames.isEmpty()) {
+            Collection<UserRoleModel> roleNames = authorizationUtility.getRoles().stream()
+                                                      .filter(role -> configuredRoleNames.contains(role.getName()))
+                                                      .collect(Collectors.toList());
+            authorizationUtility.updateUserRoles(userId, roleNames);
+        }
         userModel = userAccessor.getUser(userId).orElse(userModel);
         return convertToCustomUserRoleModel(userModel);
     }
 
+    public UserConfig updateUser(UserConfig userConfig) throws AlertFieldException {
+        //TODO when the userId is in the user config look up the user by id and update the username.
```

Comment:
Is this TODO for this PR or for a future PR?

Reply:
This is something we'll be doing in this version in another PR